src/lib/Components/Bidding/Screens/ConfirmBid.tsx

```diff
@@ -30,12 +33,121 @@ interface ConfirmBidProps extends ViewProperties {
     display: string
     cents: number
   }
+  relay?: RelayPaginationProp
+  navigator?: NavigatorIOS
+}
+
+interface ConformBidState {
+  pollCount: number
+  intervalToken: number
 }
 
-export class ConfirmBid extends React.Component<ConfirmBidProps> {
+const MAX_POLL_ATTEMPTS = 20
+
+const bidderPositionMutation = graphql`
+  mutation ConfirmBidMutation($input: BidderPositionInput!) {
+    createBidderPosition(input: $input) {
+      result {
+        position {
+          id
+        }
+        status
+        message_header
+        message_description
+        message_description_md
+      }
+    }
+  }
+`
+
+export class ConfirmBid extends React.Component<ConfirmBidProps, ConformBidState> {
+  state = {
+    pollCount: 0,
+    intervalToken: 0,
+  }
+
   onPressConditionsOfSale = () => {
     SwitchBoard.presentModalViewController(this, "/conditions-of-sale?present_modally=true")
   }
+
+  placeBid() {
+    commitMutation(this.props.relay.environment, {
+      onCompleted: (results, errors) => {
+        this.verifyBidPosition(results, errors)
+      },
+      onError: e => {
+        // TODO catch error!
+        // this.verifyAndShowBidResult(null, e)
+        console.log("error!", e, e.message)
+      },
+      mutation: bidderPositionMutation,
+      variables: {
+        input: {
+          sale_id: this.props.sale_artwork.sale.id,
+          artwork_id: this.props.sale_artwork.artwork.id,
+          max_bid_amount_cents: this.props.bid.cents,
+        },
+      },
+    })
+  }
+
+  verifyBidPosition(results, errors) {
+    const status = results.createBidderPosition.result.status
+    if (!errors && status === "SUCCESS") {
+      const positionId = results.createBidderPosition.result.position.id
+      const query = `
+        {
+          me {
+            bidder_position(id: "${positionId}") {
+              processed_at
+              is_active
+            }
+          }
+        }
+      `
+      const interval = setInterval(() => {
+        metaphysics({ query }).then(this.checkBidPosition.bind(this))
+      }, 2000)
```

Comment:
Can we double-check the desired interval? [Force uses 1000](https://github.com/artsy/force/blob/316894dc4a2ae6aa4723c730fc9635d897f82884/src/desktop/models/sale_artwork.coffee#L10) and so does Eidolon, but both of those frontends _are_ typically accessed over wifi and not 3G. In any case, let's describe why we picked that number in a comment.

Reply:
Thanks for looking it up. 👍